File: Creating a simple plugin mechanism/[VB]-Creating a simple plugin mechanism/VB/FirstPlugin/FirstPlugin.vb
Imports PluginContracts

Public Class FirstPlugin
    Implements IPlugin


    Public Sub DoSomething() Implements IPlugin.DoSomething
        System.Windows.MessageBox.Show("Do Something in First Plugin")
    End Sub

    Public ReadOnly Property Name As String Implements IPlugin.Name
        Get
            Name = "First Plugin"
        End Get
    End Property
End Class

File: Creating a simple plugin mechanism/[C#]-Creating a simple plugin mechanism/C#/FirstPlugin/FirstPlugin.cs
using PluginContracts;

namespace FirstPlugin
{
	public class FirstPlugin : IPlugin
	{
		#region IPlugin Members

		public string Name
		{
			get
			{
				return "First Plugin";
			}
		}

		public void Do()
		{
			System.Windows.MessageBox.Show("Do Something in First Plugin");
		}

		#endregion
	}
}
